### DISCUSSION TITLE
Infinite loop inside middleware

### DISCUSSION THREAD

**Discussion by voxvici** [vercel/next.js#37787] [Help] [answered]
```
import { NextResponse } from 'next/server'

export function middleware(req) {
    if (req.page.name === '/profile') {
        console.log(req.page.name);
        return NextResponse.redirect(new URL('/profile?test=a', req.url));
    }

    return NextResponse.next();
}
```

The terminal will print '/profile/' infinitely and Chrome returns ERR_TOO_MANY_REDIRECTS, is this expected behavior when adding search params inside middleware? I couldn't find anything regarding dynamic routes/search params for middleware inside docs, thanks

**Comment by alex289** [comment, marked as answer]
You are still redirecting to /profile, so the middleware will go through your if statement and redirect again. If you want this to work your if condition needs to check for query params

**Reply by voxvici** [reply]
Interesting, thank you
